update_ip/services/nfsn.py

```diff
@@ -1,64 +1,60 @@
+import logging
+
+import tld
+import pynfsn
+
 from update_ip.services.base import BaseDNSService, DNSServiceError
-try:
-    from pynfsn import pynfsn
-except ImportError:
-    raise Exception("This service requires the pynfsn package. You can find it on pypi or github")
 
-import logging
-log= logging.getLogger('update_ip.services.nfsn')
+log = logging.getLogger('update_ip.services.nfsn')
+
+ALLOW_INEXISTENT = True  # If this is false, I'll refuse to proceed if the domain record doesn't already exist
 
-ALLOW_INEXISTENT= True #If this is false, I'll refuse to proceed if the domain record doesn't already exist
 
-def split_domain(domain):
-    '''splits a complete domain into a pair of (subdomain, domain)'''
-    s= domain.split(".")
-    subdomain, domain= ".".join(s[:-2]), ".".join(s[-2:])
-    return (subdomain, domain)
-        
 class NearlyFreeSpeechService(BaseDNSService):
     name = 'NearlyfreeSpeech'
-    TTL= 300    #5 minutes of dns record ttl
-    
+    TTL = 300  # 5 minutes of dns record ttl
+
     def __init__(self, username, api_key, **kwargs):
         if not username or not api_key:
-            raise DNSServiceError('Username and api_key are required for the '
-                                 'NearlyFreeSpeech service.')
-        self.nfsn= pynfsn.NFSN(username, api_key)
+            raise DNSServiceError('Username and api_key are required for the NearlyFreeSpeech service.')
+        self.nfsn = pynfsn.NFSN(username, api_key)
 
+        # Updates the known TLD names
+        tld.utils.update_tld_names()
```

Comment:
Yes, we could - but I suspect it's going to be hard. 
Notice that `update_tld_names` writes to a file. If `tld` is installed system-wide, and `update-ip` is run as a normal (non-root) user, the function call will fail. We'd need error handling for that kind of situation. If you feel like doing all that I can review it, but it should be done in a separate PR, since this one is already very large